File: pset6/mario.py
import cs50


def main():
    """Generates a double half-pyramid of certain height."""

    # get valid height of pyramid
    while(True):
        print("Height: ", end="")
        height = cs50.get_int()
        if(height >= 0 and height <= 23):
            break

    # initialize values for first row
    hashes = 1
    spaces = height - 1

    # print pyramid row by row
    for i in range(height):
        # print left pyramid row
        print(" " * spaces, end="")
        print("#" * hashes, end="")

        # print separator of the pyramids
        print(" " * 2, end="")

        # print right pyramid row
        print("#" * hashes, end="")
        print(" " * spaces)

        # update values for next row
        spaces -= 1
        hashes += 1


if __name__ == "__main__":
    main()

File: pset1/mario.c
/**
 * Generates a double half-pyramid of certain height.
 */

#include <stdio.h>
#include <cs50.h>

void printChar(int n, char c);

int main(void)
{
    // ensure valid integer
    int height;
    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);

    // if zero height print nothing
    if (height != 0)
    {
        // generate the half pyramid row by row
        int hashes = 1;
        int spaces = height - 1;
        for (int i = 0; i < height; i++)
        {
            printChar(spaces, ' ');
            printChar(hashes, '#');
            printChar(2, ' ');
            printChar(hashes, '#');
            printf("\n");
            spaces--;
            hashes++;
        }
    }

    // success
    return 0;
}

/**
 * Prints a certain character n times.
 */
void printChar(int n, char c)
{
    for (int j = 0; j < n; j++)
    {
        printf("%c", c);
    }
}